[SR-12363] (master-next) Serialization/load-target-normalization.swift Assertion failed: (TargetInitialized && "Target not initialized!"), function isTargetWatchOSBased

|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-12363      |
   |Radar             | rdar://60097045         |
   |Original Reporter | @benlangmuir      |
   |Type              | Bug    |

<details>
  <summary>Environment</summary>

master-next


</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Compiler    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 6dbac17d9cf79358ced049f3896f48ae

  </details>





**Issue Description:**


<https://ci.swift.org/job/oss-swift-incremental-RA-osx-master-next/7054/>

    FAIL: Swift(macosx-x86_64) :: Serialization/load-target-normalization.swift (8878 of 12935)
    ******************** TEST 'Swift(macosx-x86_64) :: Serialization/load-target-normalization.swift' FAILED ********************
    Script:
    --
    <snip>
    --
    Exit Code: 1

    Command Output (stderr):
    --
    /Users/buildnode/jenkins/workspace/oss-swift-incremental-RA-osx-master-next/swift/test/Serialization/load-target-normalization.swift:18:11: error: CHECK: expected string not found in input
    // CHECK: error: could not find module 'ForeignModule'
              ^
    <stdin>:1:1: note: scanning from here
    Assertion failed: (TargetInitialized && "Target not initialized!"), function isTargetWatchOSBased, file /Users/buildnode/jenkins/workspace/oss-swift-incremental-RA-osx-master-next/llvm-project/clang/lib/Driver/ToolChains/Darwin.h, line 413.
    ^
    <stdin>:3:286: note: possible intended match here
    0. Program arguments: BUILD_DIR/bin/swiftc -frontend -target x86_64-apple-macosx10.9 -module-cache-path BUILD_DIR/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -swift-version 4 -ignore-module-source-info -typo-correction-limit 10 SOURCE_DIR/test/Serialization/load-target-normalization.swift -typecheck -I BUILD_DIR/test-macosx-x86_64/Serialization/Output/load-target-normalization.swift.tmp -parse-stdlib -Xcc -arch -Xcc i386 -target powerpc64-apple-macos10.11 
                                                                                                                                                                                                                                                                                                 ^

    --

--- a/test/Serialization/load-target-normalization.swift
+++ b/test/Serialization/load-target-normalization.swift
@@ -1,10 +1,6 @@
 // RUN: %empty-directory(%t/ForeignModule.swiftmodule)
 // RUN: touch %t/ForeignModule.swiftmodule/garbage-garbage-garbage.swiftmodule
 
-// https://github.com/apple/swift/issues/54797
-// This test crashes on 'next' branch.
-// XFAIL: asserts
-
 // Test format: We try to import ForeignModule with architectures besides
 // garbage-garbage-garbage and check the target triple listed in the error
 // message to make sure it was normalized correctly. This works in lieu of a

